internal/internal_update.go

```diff
@@ -289,12 +289,15 @@ func defaultUpdateHandler(
 		if !IsReplaying(ctx) {
 			// we don't execute update validation during replay so that
 			// validation routines can change across versions
+			defer getState(ctx).dispatcher.setIsReadOnly(false)
+			getState(ctx).dispatcher.setIsReadOnly(true)
 			if err := envInterceptor.inboundInterceptor.ValidateUpdate(ctx, &input); err != nil {
 				callbacks.Reject(err)
 				return
 			}
 		}
 		callbacks.Accept()
+		getState(ctx).dispatcher.setIsReadOnly(false)
```

Comment:
Is this line still needed?

Reply:
yes

Reply:
Ah, ok, added a comment above now that I understand the situation a bit better